src/fastify-plugins/icons.js

```diff
@@ -13,12 +13,14 @@ export default fp(iconServerPlugin, {
 const ICON_DOC_ID_STRING = T.String({ pattern: HEX_REGEX_32_BYTES })
 const PROJECT_PUBLIC_ID_STRING = T.String({ pattern: Z_BASE_32_REGEX_32_BYTES })
 
-const VALID_SIZES =
-  docSchemas.icon.properties.variants.items.properties.size.enum
-const VALID_MIME_TYPES =
-  docSchemas.icon.properties.variants.items.properties.mimeType.enum
-const VALID_PIXEL_DENSITIES =
-  docSchemas.icon.properties.variants.items.properties.pixelDensity.enum
+const VALID_SIZES = docSchemas.icon.definitions.size.enum
+const VALID_MIME_TYPES = docSchemas.icon.properties.variants.items.oneOf.map(
+  (iconType) => iconType.properties.mimeType.const
+)
+const VALID_PIXEL_DENSITIES = docSchemas.icon.properties.variants.items.oneOf
+  .filter((iconType) => iconType.properties.mimeType.const === 'image/png')
+  /** @ts-ignore using the array.filter as type guard? */
+  .map((pngIcon) => pngIcon.properties.pixelDensity.enum)
```

Comment:
To avoid the `ts-ignore`:

```suggestion
  .map((iconType) =>
    'pixelDensity' in iconType.properties
      ? iconType.properties.pixelDensity.enum
      : []
  )
  .flat()
```

This also lets us add new mime types in the future, and if they have `pixelDensity`, we'll take it.